Lib/zipfile/__init__.py

```diff
@@ -1332,12 +1332,14 @@ def __init__(self, file, mode="r", compression=ZIP_STORED, allowZip64=True,
                 "metadata_encoding is only supported for reading files")
 
         # Check if we were passed a file-like object
-        if isinstance(file, os.PathLike):
-            file = os.fspath(file)
-        if isinstance(file, str):
+        if hasattr(file, 'read') or hasattr(file, 'write'):
+            self._filePassed = 1
+            self.fp = file
+            self.filename = getattr(file, 'name', None)
+        else:
```

Comment:
What does this have to to with `tempfile`?

Reply:
When given an argument that is **both** file-like and path-like, previously `zipfile` would treat the argument as path-like and attempt to re-`open()` it. On Windows that caused test failures because the `NamedTemporaryFile` was already open. This change makes `zipfile` treat use the file-like interface in preference to the path-like interface.

Reply:
I suspect some third-party code will fall into the same trap as `zipfile` did here, and so perhaps it's not viable to make `NamedTemporaryFile` path-like.